tests/cloud/test_infer.py

```diff
@@ -0,0 +1,89 @@
+
+def test_infer(setup, mocker):
+    """
+    test_infer is a HL infer api testing function,
+    checks infer api code flow, object creations, internal api calls, internal returns.
+    ---------
+    Parameters:
+    setup: is a fixture defined in conftest.py module.
+    mocker: mocker is itself a pytest fixture, uses to mock or spy internal functions.
+    ---------
+    Ref: https://docs.pytest.org/en/7.1.x/how-to/fixtures.html
+    Ref: https://pytest-mock.readthedocs.io/en/latest/usage.html
+    """
+    ms = setup
+    get_qpc_dir_name_infer_spy = mocker.spy(QEfficient.cloud.infer,"get_qpc_dir_name_infer")
+    check_batch_size_and_num_prompts_spy = mocker.spy(QEfficient.cloud.infer,"check_batch_size_and_num_prompts")
+    load_hf_tokenizer_spy = mocker.spy(QEfficient.cloud.infer,"load_hf_tokenizer")
+    qpc_exists_spy = mocker.spy(QEfficient.cloud.infer,"qpc_exists")
+    get_onnx_model_path_spy = mocker.spy(QEfficient.cloud.infer,"get_onnx_model_path")
+    compile_spy = mocker.spy(QEfficient,"compile")
+    cloud_ai_100_exec_kv_spy = mocker.spy(QEfficient.cloud.infer,"cloud_ai_100_exec_kv")
+    except_flag = 0
+    try :
+        infer(
+                model_name = ms.model_name,
+                num_cores = ms.num_cores,
+                prompt = ms.prompt,
+                prompts_txt_file_path = ms.prompts_txt_file_path,
+                aic_enable_depth_first = ms.aic_enable_depth_first,
+                mos = ms.mos,
+                
+                hf_token = ms.hf_token,
+                batch_size = ms.batch_size,
+                prompt_len = ms.prompt_len,
+                ctx_len = ms.ctx_len,
+                mxfp6 = ms.mxfp6,
+                mxint8 = ms.mxint8,
+                device_group = ms.device_group,
+                )
+    
+    except AssertionError as e:
+        warnings.warn(e.args[0] if e.args else "error")
+        except_flag=1
+    # batch size != num of prompts
+    if except_flag==1:
+        get_qpc_dir_name_infer_spy.assert_called_once()
+        check_batch_size_and_num_prompts_spy.assert_called_once()
+        load_hf_tokenizer_spy.assert_not_called()
+        compile_spy.assert_not_called()
+        cloud_ai_100_exec_kv_spy.assert_not_called()
+    # prompt fucntion check
+    else:
+        get_qpc_dir_name_infer_spy.assert_called_once()
+        check_batch_size_and_num_prompts_spy.assert_called_once()
+        lst = check_batch_size_and_num_prompts_spy.spy_return
+        assert bool(lst) and isinstance(lst, list) and all(isinstance(elem, str) for elem in lst)
+
+        # tokenizer check
+        load_hf_tokenizer_spy.assert_called_once()
+
+        # qpc exist check
+        qpc_exists_spy.assert_called_once()
+        if qpc_exists_spy.spy_return[0] == True:
+            print("\n____________qpc exist_________________\n")
+            assert ms.qpc_dir_path() == qpc_exists_spy.spy_return[1]
+            assert os.path.isdir(ms.qpc_dir_path())

```

Comment:
add not_called checks for export and compile APIs

Reply:
added.